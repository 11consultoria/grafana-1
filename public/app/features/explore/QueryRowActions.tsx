import React from 'react';
import { Unicon } from '@grafana/ui/src/components/Icon/Unicon';

export type Props = {
  canToggleEditorModes: boolean;
  isDisabled?: boolean;
  isNotStarted: boolean;
  onClickToggleEditorMode: () => void;
  onClickToggleDisabled: () => void;
  onClickAddButton: () => void;
  onClickRemoveButton: () => void;
};

export function QueryRowActions(props: Props) {
  const {
    canToggleEditorModes,
    onClickToggleEditorMode,
    onClickToggleDisabled,
    onClickAddButton,
    onClickRemoveButton,
    isDisabled,
    isNotStarted,
  } = props;

  return (
    <div className="gf-form-inline flex-shrink-0">
      {canToggleEditorModes && (
        <div className="gf-form">
          <button
            aria-label="Edit mode button"
            className="gf-form-label gf-form-label--btn"
            onClick={onClickToggleEditorMode}
          >
            <Unicon name="pen" />
          </button>
        </div>
      )}
      <div className="gf-form">
        <button
          disabled={isNotStarted}
          className="gf-form-label gf-form-label--btn"
          onClick={onClickToggleDisabled}
          title="Disable/enable query"
        >
          <Unicon name={isDisabled ? 'eye-slash' : 'eye'} />
        </button>
      </div>
      <div className="gf-form">
        <button className="gf-form-label gf-form-label--btn" onClick={onClickAddButton} title="Add query">
          <Unicon name="plus" />
        </button>
      </div>
      <div className="gf-form">
        <button className="gf-form-label gf-form-label--btn" onClick={onClickRemoveButton} title="Remove query">
          <Unicon name="minus" />
        </button>
      </div>
    </div>
  );
}
