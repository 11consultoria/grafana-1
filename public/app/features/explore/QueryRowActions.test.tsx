import React from 'react';
import { QueryRowActions, Props } from './QueryRowActions';
import { mount, shallow } from 'enzyme';

const setup = (propOverrides?: object) => {
  const props: Props = {
    isDisabled: false,
    isNotStarted: true,
    canToggleEditorModes: true,
    onClickToggleEditorMode: () => {},
    onClickToggleDisabled: () => {},
    onClickAddButton: () => {},
    onClickRemoveButton: () => {},
  };

  Object.assign(props, propOverrides);

  const mountWrapper = mount(<QueryRowActions {...props} />);
  const shallowWrapper = shallow(<QueryRowActions {...props} />);
  return { mountWrapper, shallowWrapper };
};

describe('QueryRowActions', () => {
  it('should render component', () => {
    const wrapper = setup().shallowWrapper;
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component without editor mode', () => {
    const wrapper = setup({ canToggleEditorModes: false }).shallowWrapper;
    expect(wrapper.find({ 'aria-label': 'Edit mode button' })).toHaveLength(0);
  });
  it('should change icon to fa-eye-slash when query row result is hidden', () => {
    const wrapper = setup({ isDisabled: true }).mountWrapper;
    expect(wrapper.find('i.uil-eye-slash')).toHaveLength(1);
  });
  it('should change icon to fa-eye when query row result is not hidden', () => {
    const wrapper = setup({ isDisabled: false }).mountWrapper;
    expect(wrapper.find('i.uil-eye')).toHaveLength(1);
  });
});
