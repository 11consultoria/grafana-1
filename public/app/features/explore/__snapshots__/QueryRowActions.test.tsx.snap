// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryRowActions should render component 1`] = `
<div
  className="gf-form-inline flex-shrink-0"
>
  <div
    className="gf-form"
  >
    <button
      aria-label="Edit mode button"
      className="gf-form-label gf-form-label--btn"
      onClick={[Function]}
    >
      <Unicon
        name="pen"
      />
    </button>
  </div>
  <div
    className="gf-form"
  >
    <button
      className="gf-form-label gf-form-label--btn"
      disabled={true}
      onClick={[Function]}
      title="Disable/enable query"
    >
      <Unicon
        name="eye"
      />
    </button>
  </div>
  <div
    className="gf-form"
  >
    <button
      className="gf-form-label gf-form-label--btn"
      onClick={[Function]}
      title="Add query"
    >
      <Unicon
        name="plus"
      />
    </button>
  </div>
  <div
    className="gf-form"
  >
    <button
      className="gf-form-label gf-form-label--btn"
      onClick={[Function]}
      title="Remove query"
    >
      <Unicon
        name="minus"
      />
    </button>
  </div>
</div>
`;
